packages/tools/src/tools/annotation/SplineROITool.ts

```diff
@@ -0,0 +1,1107 @@
+      canvasPos,
+      maxDist
+    );
+
+    if (!closestControlPoint) {
+      return;
+    }
+
+    this._deleteControlPointByIndex(
+      element,
+      annotation,
+      closestControlPoint.index
+    );
+  };
+
+  /**
+   * Get a spline config merged with the default settings.
+   * @param type - Spline type (CARDINAL, CATMULLROM, LINEAR or BSPLINE)
+   * @returns Spline configuration
+   */
+  private _getSplineConfig(type: string) {
+    const { configuration: config } = this;
+    const splineConfigs = config.spline.configuration;
+
+    return Object.assign({ type }, DEFAULT_SPLINE_CONFIG, splineConfigs[type]);
+  }
+
+  private _updateSplineScale(spline: ISpline, annotation: SplineROIAnnotation) {
+    const splineType = annotation.data.spline.type;
+    const splineConfig = this._getSplineConfig(splineType);
+
+    if (
+      !(spline instanceof CardinalSpline) ||
+      spline.fixedScale ||
+      splineConfig.scale === undefined ||
+      spline.scale === splineConfig.scale
+    ) {
+      return;
+    }
+
+    spline.scale = splineConfig.scale;
+    annotation.invalidated = true;
+  }
+
+  private _createOrUpdateAnnotationSpline = (
+    element: HTMLDivElement,
+    annotation: SplineROIAnnotation
+  ): ISpline => {
+    const enabledElement = getEnabledElement(element);
+    const { viewport } = enabledElement;
+    const { worldToCanvas } = viewport;
+    const splineType = annotation.data.spline.type;
+    const splineConfig = this._getSplineConfig(splineType);
+    let spline = this.splineInstancesCache.get(annotation);
+
+    if (!spline) {
+      spline = new splineConfig.Class();
+      this.splineInstancesCache.set(annotation, spline);
+    }
+
+    const { data } = annotation;
+    const worldPoints = data.handles.points;
+    const canvasPoints = worldPoints.map(worldToCanvas);
+
+    spline.setControlPoints(canvasPoints);
+    spline.closed = !!data.spline?.closed;
+
+    // Update spline resolution in case it has changed
+    if (spline.resolution !== splineConfig.resolution) {
+      spline.resolution = parseInt(splineConfig.resolution);
+      annotation.invalidated = true;
+    }
+
+    // Update Cardinal spline scale in case it has changed
+    if (
+      spline instanceof CardinalSpline &&
+      !spline.fixedScale &&
+      splineConfig.scale !== undefined &&
+      spline.scale !== splineConfig.scale
+    ) {
+      spline.scale = splineConfig.scale;
+      annotation.invalidated = true;
+    }
+
+    return spline;
+  };
+
+  private _getAnnotationSpline = (
+    element: HTMLDivElement,
+    annotation: SplineROIAnnotation
+  ): ISpline => {
+    let spline = this.splineInstancesCache.get(annotation);
+
+    if (!spline) {
+      spline = this._createOrUpdateAnnotationSpline(element, annotation);
+    }
+
+    return spline;
+  };
+
+  private _deleteAnnotationSpline = (annotation: SplineROIAnnotation) => {
+    this.splineInstancesCache.delete(annotation);
+  };
+
```

Comment:
This is a side effect of having it outside the annotation manager, ideally remove annotation should remove its spline instances

Reply:
Later, we can extract the serializable pieces and reinstantiate the class.